dsls/sql/src/main/java/org/apache/beam/dsls/sql/interpreter/operator/BeamSqlPrimitive.java

```diff
@@ -67,34 +67,34 @@ public T getValue() {
   }
 
   public long getLong() {
-    return (Long) getValue();
+    return Long.valueOf(getString());
```

Comment:
if it is of type short/long and has to be converted to double then in that case it will fail to convert.